node/network/collator-protocol/src/collator_side.rs

```diff
@@ -412,8 +445,9 @@ async fn advertise_collation(
 		.map(|g| g.should_advertise_to(&peer))
 		.unwrap_or(false);
 
-	if !state.collations.contains_key(&relay_parent) || !should_advertise {
-		return;
+	match (state.collations.get_mut(&relay_parent), should_advertise) {
+		(None, _) | (_, false) => return,
```

Comment:
Nit: Could we spread these arms with a `trace`  why we exit? This could give us a clue why a collation wasn't advertised